content/client-server-api/modules/end_to_end_encryption.md

```diff
@@ -1596,6 +1624,13 @@ that they can decrypt future messages encrypted using this session. An
 `m.room_key` events sent by other devices in order to decrypt their
 messages.
 
+When a client is updating a session key in its store, the client MUST ensure:
+
+* that the updated session data comes from a trusted source, where a trusted source
+  could be the signatures on the session or comes from the user's session key backup;
+  and
+* that the new session key has a lower message index than the existing session key.
```

Comment:
```suggestion
When a client is updating a Megolm session (room key) in its store, the client MUST ensure:

* that the updated session data comes from a trusted source, where a trusted source
  could be the signatures on the session or comes from the user's session key backup;
  and
* that the new session has a lower message index than the existing session.
```

In order to:

1. Be explicit about this being a Megolm session. The word "session" is overloaded.
2. Use "session" instead of "session key", since there's really no sane difference between these terms. We usually say "session" or "room key", and these are synonyms on the Matrix level.

This doesn't really fix the existing terminology mess and that should be cleaned up separately, but at least we won't introduce new ambiguities.

Reply:
applied manually